api/v1beta1/hetznercluster_webhook.go

```diff
@@ -109,6 +149,55 @@ func (r *HetznerCluster) ValidateCreate() (admission.Warnings, error) {
 		if err := isNetworkZoneSameForAllRegions(r.Spec.ControlPlaneRegions, nil); err != nil {
 			allErrs = append(allErrs, err)
 		}
+	} else {
+		// If ID is given check that all other network settings are empty.
+		if r.Spec.HCloudNetwork.ID != nil {
+			if errs := areCIDRsAndNetworkZoneEmpty(r.Spec.HCloudNetwork); errs != nil {
+				allErrs = append(allErrs, errs...)
+			}
+		} else {
+			// If no ID is given check the other network settings for valid entries.
+			if r.Spec.HCloudNetwork.NetworkZone != nil {
```

Comment:
what is supposed to happen if this is nil? 

I just wanted to say that we should again avoid the nested if statements if possible and that we can do that here. 

Currently, we don't handle the case where NetworkZone is nil at all. Shouldn't we though? Isn't that a problem? 

Anyway, let's avoid the nested if statement and if you say we should, then let's handle the case where NetworkZone is nil

Reply:
I agree that we should probably append an error when NetworkZone is nil. But still I'm struggling to see how to really circumvent the nested statements because the method's idea seems to be to aggregate all errors that might occur instead of returning early. An early return would then potentially miss other validation errors that might happen afterwards. But probably I'm just missing a detail.

Reply:
I wanted to avoid the large code block for "NetworkZone != nil". However, I think you are right that this is not really possible, as we cannot just return at that point. 

Therefore, I would say let's add a handling of "NetworkZone == nil" and otherwise don't change anything. Does that make sense to you?

Reply:
Sure! And yeah, it's not the most pretty thing 😅.
Regarding the "NetworkZone == nil". So do you mean to only handle the nil case and delete the validation of the network zones or handle both cases? In that vein, should I also handle the nil case for cidrBlock and subnetCIDRBlock as well?

Reply:
you did that already right? Do you think we miss any important case right now, or is this done?

Reply:
Yep. Did this as well.